Dynatrace/dynatrace-operator
# Adds serverPort for webhook in helm chart, and adds the same option for metricsPort and healthProbePort

Adds serverPort for webhook in helm chart, and adds the same option for metricsPort and healthProbePort


For setups using EKS with a third-party CNI (e.g. Cilium) in overlay mode, the webhook needs to run with `hostNetwork: true`. 
This presents a potential issue with conflicting ports. #2624 addressed this on the webhook itself, but did not add the option for the helm chart. This PR adds this setting to the helm chart, and expands the functionality by allowing to set healthProbePort and metricsPort. 


The code changes are:
- Add environment variables to the deployment in helm
- Modify the webhook manager.go code to use the environment variables for metricsPort and healthProbePort if provided (same as was done in #2624 for serverPort)
- Modify tests to include all custom ports
- Modify values file to include the new settings


These are the steps to validate the PR:
- Create an EKS cluster with Cilium running on overlay mode
- Create a deployment using hostNetwork and open one of the ports used by the webhook and a number of replicas of at least equals to the number of nodes
- Deploy the dynatrace-operator helm chart with default ports
- Check that the dynatrace-webhook is pending due to unavailable ports
- Specify new, non-conflicting ports for the webhook
- Check that the pods now start as expected
